src/main/java/BeeBot.java

```diff
@@ -0,0 +1,168 @@
+ * from/to a specified file.
+ * </p>
+ */
+public class BeeBot {
+    private static Storage storage;
+    private static ArrayList<Task> taskList;
+    private final String FILEPATH;
+
+    /**
+     * Constructs a {@code BeeBot} instance with the specified file path.
+     * <p>
+     * Initializes the storage and attempts to load the task list from the provided file path. If loading fails,
+     * initializes an empty task list.
+     * </p>
+     *
+     * @param filePath the path to the file used for loading and saving tasks
+     */
+    public BeeBot(String filePath) {
+        this.FILEPATH = filePath;
+        storage = new Storage();
+        Ui ui = new Ui();
+        try {
+            taskList = storage.loadTaskListFromFile(filePath);
+        } catch (Exception e) {
+            System.out.println("Error loading file: " + e.getMessage());
+            taskList = new ArrayList<>();
+        }
+    }
+
+    /**
+     * The entry point of the application.
+     * <p>
+     * Creates an instance of {@code BeeBot} with a predefined file path and starts the main loop to handle user input.
+     * </p>
+     *
+     * @param args command-line arguments (not used)
+     */
+    public static void main(String[] args) {
+        new BeeBot("/Users/beetee/Documents/NUS/CS/Y2S1/CS2103T/ip/data/BeeBot.txt").run();
+    }
+
+    /**
+     * Starts the main loop for handling user input.
+     * <p>
+     * Reads commands from the user, processes them, and updates the task list accordingly. It handles various commands,
+     * such as adding, marking, and deleting tasks, and interacts with the {@code Ui} and {@code Parser} classes.
+     * </p>
+     */
+    public void run() {
+        boolean exit_status = false;
+        while (!exit_status) {
+            Scanner scanner = new Scanner(System.in);
+            String input = scanner.nextLine();
+            String[] parts = input.split(" ");
+            String cmd = parts[0];
+
+            try {
+                switch (cmd) {
+                case "bye":
+                    Ui.exit();
+                    exit_status = true;
+                    break;
+                case "list":
+                    int size = taskList.size();
+                    String listStr = "";
+                    for (int i = 0; i < size; i++) {
+                        int num = i + 1;
+                        listStr += (num + "." + taskList.get(i).toString());
+                    }
+                    Parser.speak(listStr);
+                    break;
+                case "mark":
+                    int taskNum = Integer.parseInt(parts[1]);
+                    Task currTask = Parser.getTask(taskList, taskNum);
+                    currTask.markAsDone();
+                    Parser.speak("Nice! I've marked this task as done:\n" + currTask.toString());
+                    break;
+                case "unmark":
+                    int taskNum2 = Integer.parseInt(parts[1]);
```

Comment:
is there a better name other than taskNum2?

Reply:
yes i have changed it! thanks for the comment